docker/docker_image.go

```diff
@@ -19,7 +19,17 @@ type Image struct {
 // NewDockerImage returns a new Image interface type after setting up
 // a client to the registry hosting the given image.
 func NewDockerImage(img, certPath string, tlsVerify bool) (types.Image, error) {
-	s, err := newDockerImageSource(img, certPath, tlsVerify)
+	ref, err := parseReference("//" + img)
+	if err != nil {
+		return nil, err
+	}
+	return newDockerImage(ref, certPath, tlsVerify)
+}
```

Comment:
This PR adds these `parseReference`+`new${x}Image${y}` helpers all over the place. I am not sure that this makes much sense. Yes, it is nice not to have to do any parsing in the caller… OTOH there are now so many similar and redundant ways to get from a string to an `types.Image$something`.

Arguably anyone starting with a string should go through https://github.com/projectatomic/skopeo/pull/138/files#diff-6179b09a11bd9337d53f297f715ac27dR13 (or an equivalent in this repo); and any caller knowledgeable enoug about a transport to directoy call one of these functions might be knowledgeable enough to give it a fully parsed specification (`reference.Named` + tag for Docker, `namespace`, `stream`, `tag` for OpenShift) etc.

I’ll take a look at this tomorrow with fresh eyes, comments very much apprecireated.